src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -567,17 +568,25 @@ internal void ReadToken(ReadOnlyMemory<char> encodedTokenMemory)
 
         internal JsonClaimSet CreateClaimSet(ReadOnlySpan<char> strSpan, int startIndex, int length, bool createHeaderClaimSet)
         {
-            int outputSize = Base64UrlEncoding.ValidateAndGetOutputSize(strSpan, startIndex, length);
+            ReadOnlySpan<char> slice = Base64UrlEncoding.Validate(strSpan, startIndex, length);
+            int outputsize = Base64Url.GetMaxDecodedLength(length);
+
+            byte[] rented = null;
+
+            const int MaxStackallocThreshold = 256;
+            Span<byte> destination = outputsize <= MaxStackallocThreshold
+                ? stackalloc byte[outputsize]
+                : (rented = ArrayPool<byte>.Shared.Rent(outputsize));
```

Comment:
Most likely not. This was copied from the internal implementation of the `Base64Url` API, and since this is all existing code being updated no new tests were added.

Reply:
I think this is ok. This will get tested by all other tests that create a JsonWebToken.